modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/networkapi/NetworkApiService.scala

```diff
@@ -0,0 +1,62 @@
+package org.tessellation.rosetta.domain.networkapi
+
+import cats.data.EitherT
+import cats.effect.Async
+
+import org.tessellation.BuildInfo
+import org.tessellation.rosetta.domain.NetworkIdentifier
+import org.tessellation.rosetta.domain.error.{LatestSnapshotNotFound, NetworkApiError}
+import org.tessellation.rosetta.domain.networkapi.model.options._
+import org.tessellation.rosetta.domain.networkapi.model.status._
+import org.tessellation.schema.{GlobalSnapshot, SnapshotOrdinal}
+import org.tessellation.security.Hashed
+import org.tessellation.security.hex.Hex
+
+trait NetworkApiService[F[_]] {
+  def list: EitherT[F, NetworkApiError, List[NetworkIdentifier]]
+  def options: EitherT[F, NetworkApiError, NetworkApiOptions]
+  def status: EitherT[F, NetworkApiError, NetworkStatusResponse]
+}
+
+object NetworkApiService {
+  private val rosettaVersion = "1.4.14" // TODO get from a config?
+  private val version = Version(rosettaVersion = rosettaVersion, nodeVersion = BuildInfo.version)
+
+  def make[F[_]: Async](
+    getLastSnapshot: () => F[Option[Hashed[GlobalSnapshot]]],
+    getGenesisSnapshot: () => F[GlobalSnapshot],
+    getTimestampForSnapshot: SnapshotOrdinal => F[Long]
+  ): NetworkApiService[F] = new NetworkApiService[F] {
+    def list: EitherT[F, NetworkApiError, List[NetworkIdentifier]] =
+      EitherT.pure(List(NetworkIdentifier.Mainnet, NetworkIdentifier.Testnet))
+
+    def options: EitherT[F, NetworkApiError, NetworkApiOptions] =
+      EitherT.pure(NetworkApiOptions(version, Allow.default))
+
+    def status: EitherT[F, NetworkApiError, NetworkStatusResponse] = for {
+      lastSnapshot <- EitherT.fromOptionF[F, NetworkApiError, Hashed[GlobalSnapshot]](
```

Comment:
in general, when you need to lift each and every operation to `EitherT` it may be beneficial to chain operations and lift to EitherT once at the very end

Reply:
@marcinwadon
I don't understand. Only 2 of the 3 uses are lifted and that's so the first one can fail fast. As I understand it that's the point of using a monad transformer.

Reply:
in this case you have one operation that is lifted to EitherT because it returns Option (and you want to fail fast).
but then, you have some other operations that you need to lift into Either (each one). if you would have more of them, it may be simpler to just chain them and use `EitherT.liftF { ...chainedOperations... }` once. something like that (or equivalent using for comp):
![image](https://user-images.githubusercontent.com/4959931/236476689-427d44ed-5678-4633-a93e-f933f959f188.png)

Reply:
Seems to me "simpler" is subjective. If you insist I'll change it to that but I don't see value in it.

Reply:
No problem. I've just written that if you'd have a more complex case with more operations. It's fine for now

Reply:
ok, thank you Marcin. I will continue to look for simpler implementations and only lift things into the EitherT if it feels like it makes the code easier to understand.